src/minswap/datums/order-datum.ts

```diff
@@ -22,34 +23,51 @@ export class MinswapOrderDatumDecoder implements Decodable<IMinswapOrderDatum> {
   static new = (network: Network) => new MinswapOrderDatumDecoder(network);
 
   decode(cborHex: string): IMinswapOrderDatum {
-    const pd = PlutusData.from_bytes(fromHex(cborHex));
-    const cpd = pd.as_constr_plutus_data();
-    if (!cpd) throw new Error('Invalid constructor plutus data for order datum');
-    const fields = cpd.data();
-    if (fields.len() !== 6) throw new Error(`Expected exactly 6 fields for order datum, received: ${fields.len()}`);
+    const mfs = new ManagedFreeableScope();
+    const fields = PlutusData.from_bytes(fromHex(cborHex)).as_constr_plutus_data()?.data();
```

Comment:
You need to call `mfs.manage` for every CSL object you get in javascript. This is only managing the last object you get in the expression, these are missing:
```suggestion
    const fields = mfs.manage(mfs.manage(PlutusData.from_bytes(fromHex(cborHex))).as_constr_plutus_data())?.data();
```
This applies to all other CSL usages too. Not just this specific case. All objects that are returned from CSL that have `free()` must be manually freed. Including:
- collections (the ones you use like `.get(i)`)
- arguments passed to CSL (like 99%, with a few exceptions), because CSL typically clones the objects it receives.
- primitive CSL types, like `BigNum`, `Integer` etc

Reply:
Thanks 👍